backend/src/modules/doubles-tournament/core/application/event/CallMatchWhenTournamentTableWasReleased.ts

```diff
@@ -0,0 +1,42 @@
+import { EventHandler } from '../../../../../shared/core/application/event/EventHandler';
+import { CommandPublisher } from '../../../../../shared/core/application/command/CommandBus';
+import { MatchesQueueRepository } from '../MatchesQueueRepository';
+import { TablesQueueRepository } from '../TablesQueueRepository';
+import { TournamentTableWasReleased } from '../../../../tournament-tables/core/domain/event/TournamentTableWasReleased';
+import { CallMatch } from '../command/CallMatch';
+
+export class CallMatchWhenTournamentTableWasReleased implements EventHandler<TournamentTableWasReleased> {
+  private readonly commandPublisher: CommandPublisher;
+  private readonly matchesQueueRepository: MatchesQueueRepository;
+  private readonly tablesQueueRepository: TablesQueueRepository;
+
+  constructor(
+    commandPublisher: CommandPublisher,
+    matchesQueueRepository: MatchesQueueRepository,
+    tablesQueueRepository: TablesQueueRepository,
+  ) {
+    this.commandPublisher = commandPublisher;
+    this.matchesQueueRepository = matchesQueueRepository;
+    this.tablesQueueRepository = tablesQueueRepository;
+  }
+
+  async handle(event: TournamentTableWasReleased): Promise<void> {
```

Comment:
Hmmm... chyba to jest OK, ale nie wiem czy nie porsciej byłoby zrobić coś co zbierałoby te 2 kolejki (meczy i stołów) i po prostu po każdej operacji wpływającej na możliwość wywołania - sprawdzać czy jest i wolny stół w kolejce i mecz gotowy do wywołania.

Ale to tez chyba jest w porządku :) Ale zdaje mi się, że bardziej komplikuje.

Reply:
Najpierw kombinowałam z jednym handlerem, który przyjmował eventy dwóch typów, ale doszłam do wniosku, że chyba lepiej to rozdzielić.

Reply:
Handlery zawsze mogą być dwa, ale potem wywoływać jakąś funkcję w tym samym obiekcie :) Ale skoro działa dobrze to może zostać. Ale jeszcze nie wiem, czy działa :D Wydaje mi się trochę skomplikowane przez takie rozdzielenie.